### DISCUSSION TITLE
Error when running a modified OpenFAST model although compilation was successful

### DISCUSSION THREAD

**Discussion by relhamoud** [OpenFAST/openfast#2465] [Q&A] [answered]
Hello everyone,

I modified the source code of openFAST more particularly the StrucCtrl.f90 in order to account for an external force in the aim of an active structural control using an active TMD. The compilation was successful. However, when running openFAST as usual the following error occurs:

**Running ServoDyn.
forrtl: severe (157): Program Exception - access violation
Image              PC                Routine            Line        Source
openfast_x64_ne_T  00007FF67A7FBB70  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF67A05BA7E  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF67A075147  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF6792F1323  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF6792FA24E  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF679261294  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF67B8D3CDB  Unknown               Unknown  Unknown
openfast_x64_ne_T  00007FF67BBD4200  Unknown               Unknown  Unknown
KERNEL32.DLL       00007FF94BD87974  Unknown               Unknown  Unknown
ntdll.dll          00007FF94C66A271  Unknown               Unknown  Unknown
forrtl: severe (28): CLOSE error, unit 7, file "Unknown"**

Could you help me please in understanding and fixing the error?

Thank you in advance.

Best Regards,

Riad

**Comment by jjonkman** [comment, marked as answer]
Dear @relhamoud,

A program exception access violation error typically occurs when you are trying to write to a memory location that is inaccessible or outside of the process address space, for example, writing to an element of an array that outside the bounds of the array.

Best regards,
